sources/packages/backend/apps/api/src/route-controllers/application-restriction-bypass/_tests_/e2e/application-restriction-bypass.aest.controller.bypassRestriction.e2e-spec.ts

```diff
@@ -0,0 +1,275 @@
+      .send(payload)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        message:
+          "Cannot create a bypass when there is an active bypass for the same active student restriction.",
+        errorType: ACTIVE_BYPASS_FOR_STUDENT_RESTRICTION_ALREADY_EXISTS,
+      });
+  });
+
+  it("Should throw an HTTP error while creating a bypass when student restriction is not active.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, undefined, {
+      offeringIntensity: OfferingIntensity.partTime,
+    });
+
+    const studentRestriction = await saveFakeStudentRestriction(
+      db.dataSource,
+      {
+        student: application.student,
+        restriction: ptssrRestriction,
+        resolutionNote: null,
+        creator: sharedMinistryUser,
+      },
+      {
+        isActive: false,
+      },
+    );
+
+    const payload = {
+      applicationId: application.id,
+      studentRestrictionId: studentRestriction.id,
+      bypassBehavior: RestrictionBypassBehaviors.NextDisbursementOnly,
+      note: "test note",
+    };
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        message:
+          "Cannot create a bypass when student restriction is not active.",
+        errorType: STUDENT_RESTRICTION_IS_NOT_ACTIVE,
+      });
+  });
+
+  it("Should throw an HTTP error while creating a bypass when the student application is in draft.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, undefined, {
+      offeringIntensity: OfferingIntensity.partTime,
+      applicationStatus: ApplicationStatus.Draft,
+    });
+
+    const studentRestriction = await saveFakeStudentRestriction(db.dataSource, {
+      student: application.student,
+      restriction: ptssrRestriction,
+      resolutionNote: null,
+      creator: sharedMinistryUser,
+    });
+
+    const payload = {
+      applicationId: application.id,
+      studentRestrictionId: studentRestriction.id,
+      bypassBehavior: RestrictionBypassBehaviors.NextDisbursementOnly,
+      note: "test note",
+    };
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        message: "Cannot create a bypass when application is in invalid state.",
+        errorType:
+          APPLICATION_IN_INVALID_STATE_FOR_APPLICATION_RESTRICTION_BYPASS_CREATION,
+      });
+  });
+
+  it("Should throw an HTTP error while creating a bypass when the student application is in cancelled.", async () => {
```

Comment:
There is a minor typo here: "application is in cancelled" should be "application is cancelled".